core/src/main/java/info/rmapproject/core/idservice/HttpArkIdService.java

```diff
@@ -0,0 +1,395 @@
+ */
+
+package info.rmapproject.core.idservice;
+
+import edu.ucsb.nceas.ezid.EZIDClient;
+import edu.ucsb.nceas.ezid.EZIDException;
+import org.h2.mvstore.MVMap;
+import org.h2.mvstore.MVStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.URI;
+
+public class HttpArkIdService implements IdService {
+
+    /** The log. */
+    private static final Logger log = LoggerFactory.getLogger(HttpUrlIdService.class);
+
+    /**  Wait time to retry when ID retrieval unsuccessful (5 seconds). */
+    private static final int RETRY_WAIT_TIME = 5000;
+
+    /** The HTTP ID service url. */
+    private String serviceUrl = "";
+
+    /** The ID prefix to be added. */
+    private String idPrefix = "";
+
+    /** Set a maximum retry attempts value if you want to cap the property setting. -1 means no maximum */
+    private int maxRetryAttempts = -1;
+
+    /** User name if using Basic Auth. */
+    private String userName = "";
+
+    /** User pwd if using Basic Auth. */
+    private String userPassword = "";
+
+    /** Length of ID to validate against. */
+    private int idLength = -1;
+
+    /** String regex to validate an ID against. */
+    private String idRegex = "";
+
+    /**file for persistent reserved ID storage **/
+    private String idStoreFile = "/tmp/idStoreFile";
+    private static final String DATA = "ezidData";
+    /** the MVMap for storing the reserved EZIDs **/
+    private MVMap<Integer, String> ezids;
+    private int maxStoreSize=200;
+    /** indicates whether we have a process running which is replenishing the cache **/
+    private boolean replenishingCache = false;
+
+
+    /**
+     * Instantiates a new ARK ID service.
+     */
+    public HttpArkIdService() {
+
+    }
+
+    /* (non-Javadoc)
+     * @see info.rmapproject.core.idservice.IdService#isValidId(java.net.URI)
+     */
+    @Override
+    public boolean isValidId(URI id) throws Exception {
+        return isValidId(id.toASCIIString());
+    }
+
+
+    /**
+     * Check the string value of an ID is valid by checking it matches a regex and is the right length
+     *
+     * @param id the id
+     * @return boolean
+     * @throws Exception the exception
+     */
+    private boolean isValidId(String id) throws Exception {
+        boolean isValid = true;
+        if (idRegex!=null && idRegex.length()>0){
+            isValid = id.matches(idRegex);
+        }
+        if (isValid && idLength>0) {
+            isValid = (id.length()==idLength);
+        }
+        return isValid;
+    }
+
+    /* (non-Javadoc)
+     * @see info.rmapproject.core.idservice.IdService#createId()
+     */
+    public URI createId() throws Exception {
+        try {
+            return new URI(getEzid());
+        } catch (Exception e) {
+            throw new Exception("Failed to create a new ID.", e);
+        }
+    }
+
+    private synchronized String getEzid() throws Exception {
+        String id;
+        MVStore mvs = MVStore.open(idStoreFile);
```

Comment:
Does it matter if `MVStore` is not closed?  For example, if an exception is thrown below, the store won't be closed.  Is that a problem (not familiar with how `MVStore` manages resources).